gazelle_haskell_modules/rule_generation.go

```diff
@@ -398,17 +398,29 @@ type ModuleImport struct {
 	ModuleName  string
 }
 
+type TempModuleImport struct {
+	IsSourceImported bool
+	ModuleName       []string
+}
+
 func (moduleImport *ModuleImport) UnmarshalJSON(data []byte) error {
-	var aux []string
-	if err := json.Unmarshal(data, &aux); err != nil {
+	var tempMod TempModuleImport
+	if err := json.Unmarshal(data, &tempMod); err != nil {
 		return err
 	}
+	aux := tempMod.ModuleName
 	pkgName := ""
 	if len(aux) > 1 {
 		pkgName = aux[0]
 	}
+	suffix := ""
+	// We add the suffix `.hs-boot` to the module name,
+	// when this module comes from an `hs-boot` file.
+	if tempMod.IsSourceImported {
+		suffix = ".hs-boot"
```

Comment:
I think you should factor out `".hs-boot"` in a similar manner to the other "global constants", e.g. 
```golang
const PRIVATE_ATTR_MODULE_LABELS = "module_labels"
```